src/charm.py

```diff
@@ -607,6 +668,8 @@ def _on_leader_elected(self, event: LeaderElectedEvent) -> None:
 
     def _on_pebble_ready(self, event: PebbleReadyEvent) -> None:
         """Event Handler for pebble ready event."""
+        self._patch_statefulset()
```

Comment:
Just have a small question: any reason we patch the StatefulSet in the `x-pebble-ready` event handle?

Reply:
AFAIK it is common practice to do it in that event.

I think that the reason the `pebble-ready` event is considered ideal for this is that we want all our charms to patch the statefulSet on the same event, so it should not be a "custom" event (like database created). Also we want it to happen early, as it causes the pod to restart. IMO the only other event where it makes sense to patch the statefulSe is `install`. According to the [juju docs](https://juju.is/docs/sdk/install-event#heading--emission-sequence), the `install` event seems like a fine place to patch the statefulSet as well, BUT it is not called when the charm is upgraded so we would have to patch the statefulSet on both `install` and `upgrade-charm`. Tbh I have no problem with that but I decided to follow what (I think) is common practice.